ui/lib/kubernetes/addon/components/page/credentials.hbs

```diff
@@ -0,0 +1,144 @@
+        @buttonType="button"
+        @success={{action (set-flash-message "Service account name copied")}}
+      >
+        <Icon class="has-text-grey-light" @name="clipboard-copy" aria-label="Copy" />
+        <span class="has-font-weight-normal title is-7">{{this.credentials.service_account_name}}</span>
+      </CopyButton>
+    </InfoTableRow>
+
+    <InfoTableRow @label="Lease expiry" @value={{date-format this.leaseExpiry "MMMM do yyyy, h:mm:ss a"}} />
+    <InfoTableRow @label="lease_id" @value={{this.credentials.lease_id}} />
+    <InfoTableRow />
+  </div>
+
+  <div class="has-top-margin-l">
+    <button class="button is-primary" type="button" {{on "click" this.cancel}}>
+      Done
+    </button>
+  </div>
+{{else}}
+  <div class="box is-sideless is-fullwidth is-marginless" data-test-generate-credentials>
+    <p>This will generate credentials using the role <span class="is-font-mono">rolename</span>.</p>
+
+    {{#if this.error}}
+      <MessageError class="has-top-margin-l" @errorMessage={{this.error}} />
+    {{/if}}
+
+    <div class="field has-top-margin-l">
+      <label for="token" class="is-label">Kubernetes namespace</label>
+      <div class="has-text-grey is-size-8 has-bottom-margin-xs">
+        The namespace in which to generate the credentials.
+      </div>
+      <Input
+        @type="text"
+        @value={{this.kubernetesNamespace}}
+        class="input"
+        {{on "input" this.setKubernetesNamespace}}
+        data-test-kubernetes-namespace
+      />
+
+      <div class="has-top-margin-l has-bottom-margin-l">
+        <Toggle
+          @status="success"
+          @size="small"
+          @checked={{this.clusterRoleBinding}}
+          @onChange={{(toggle-action "clusterRoleBinding" this)}}
+          data-test-kubernetes-clusterRoleBinding
+        >
+          <h3 class="title is-7 is-marginless">ClusterRoleBinding</h3>
+          <div class="description has-text-grey">
+            <span>
+              Generate a ClusterRoleBinding to grant permissions across the whole cluster instead of within a namespace. This
+              requires the Vault role to have kubernetes_role_type set to ClusterRole.
+            </span>
+          </div>
+        </Toggle>
+      </div>
+
+      <TtlPicker2
+        class="has-top-margin-l has-bottom-margin-m"
+        @initialEnabled={{false}}
+        @label="Time-to-Live (TTL)"
+        @onChange={{this.updateTtl}}
+        @helperTextDisabled="The TTL of the generated Kubernetes service account token. Defaults to the role's default TTL, or the default system TTL."
+      />
+    </div>
+  </div>
+
+  <div class="has-top-margin-l">
+    <button
+      class="button is-primary"
+      type="button"
+      {{on "click" (perform this.fetchCredentials)}}
+      data-test-generate-credentials-button
+    >
+      Generate credentials
+    </button>
+    <button class="button has-left-margin-xs" type="button" {{on "click" this.cancel}}>
+      Back
+    </button>
```

Comment:
Since you are using a concurrency task, these buttons could be disabled when `this.fetchCredentials.isRunning`.

Reply:
Thanks for catching that!